Added doc on keyword break

dorfsmay opened the pull request:
RE: #34601

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @cramertj (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

rust-highfive commented:
The job `x86_64-gnu-llvm-6.0` of your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=11049&view=logs&jobId=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e) ([pretty log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/11049/logs/49), [raw log](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/11049/logs/49)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2019-10-18T05:38:55.9143774Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-10-18T05:38:55.9373483Z ##[command]git config gc.auto 0
2019-10-18T05:38:55.9447235Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-10-18T05:38:55.9500633Z ##[command]git config --get-all http.proxy
2019-10-18T05:38:55.9637685Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/65544/merge:refs/remotes/pull/65544/merge
---
2019-10-18T06:38:55.8168130Z .................................................................................................... 1600/9196
2019-10-18T06:39:00.7229206Z .................................................................................................... 1700/9196
2019-10-18T06:39:13.1223433Z .............................i...............i...................................................... 1800/9196
2019-10-18T06:39:20.4960434Z .................................................................................................... 1900/9196
2019-10-18T06:39:34.5850233Z ...................iiiii............................................................................ 2000/9196
2019-10-18T06:39:44.6321739Z .................................................................................................... 2200/9196
2019-10-18T06:39:47.0990388Z .................................................................................................... 2300/9196
2019-10-18T06:39:52.2355457Z .................................................................................................... 2400/9196
2019-10-18T06:40:13.4783043Z .................................................................................................... 2500/9196
---
2019-10-18T06:43:01.7595508Z .......................i..............i............................................................. 4800/9196
2019-10-18T06:43:12.8866426Z .................................................................................................... 4900/9196
2019-10-18T06:43:18.9791330Z .................................................................................................... 5000/9196
2019-10-18T06:43:28.7902556Z .................................................................................................... 5100/9196
2019-10-18T06:43:34.9449232Z ......................ii.ii......................................................................... 5200/9196
2019-10-18T06:43:45.0431406Z .................................................................................................... 5400/9196
2019-10-18T06:43:54.7259427Z ........................................................................................i........... 5500/9196
2019-10-18T06:44:02.4423247Z .................................................................................................... 5600/9196
2019-10-18T06:44:07.2635631Z .................................................................................................... 5700/9196
2019-10-18T06:44:07.2635631Z .................................................................................................... 5700/9196
2019-10-18T06:44:17.6748466Z .....................................................................................ii...i..ii..... 5800/9196
2019-10-18T06:44:42.7462493Z .................................................................................................... 6000/9196
2019-10-18T06:44:51.9685986Z .................................................................................................... 6100/9196
2019-10-18T06:44:58.1747118Z .................................................................................................... 6200/9196
2019-10-18T06:44:58.1747118Z .................................................................................................... 6200/9196
2019-10-18T06:45:11.5625590Z .......i..ii........................................................................................ 6300/9196
2019-10-18T06:45:30.3876997Z ...................................................................i................................ 6500/9196
2019-10-18T06:45:32.5167259Z .................................................................................................... 6600/9196
2019-10-18T06:45:34.9217759Z .........................................i.......................................................... 6700/9196
2019-10-18T06:45:38.6369415Z .................................................................................................... 6800/9196
---
2019-10-18T06:50:00.2960496Z  finished in 5.120
2019-10-18T06:50:00.3135800Z Check compiletest suite=codegen mode=codegen (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-18T06:50:00.4651862Z 
2019-10-18T06:50:00.4652097Z running 153 tests
2019-10-18T06:50:03.5640977Z i....iii......iii..iiii...i.............................i..i..................i....i...........ii.i. 100/153
2019-10-18T06:50:05.4841694Z i..iiii..............i.........iii.i.........ii......
2019-10-18T06:50:05.4843336Z 
2019-10-18T06:50:05.4844960Z  finished in 5.170
2019-10-18T06:50:05.5014847Z Check compiletest suite=codegen-units mode=codegen-units (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-18T06:50:06.0719406Z 
---
2019-10-18T06:50:07.6174754Z  finished in 2.116
2019-10-18T06:50:07.6372782Z Check compiletest suite=assembly mode=assembly (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-18T06:50:07.7891754Z 
2019-10-18T06:50:07.7892301Z running 9 tests
2019-10-18T06:50:07.7893187Z iiiiiiiii
2019-10-18T06:50:07.7893910Z 
2019-10-18T06:50:07.7896954Z  finished in 0.152
2019-10-18T06:50:07.8075971Z Check compiletest suite=incremental mode=incremental (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-18T06:50:07.9623749Z 
---
2019-10-18T06:50:25.8524268Z  finished in 18.044
2019-10-18T06:50:25.8728997Z Check compiletest suite=debuginfo mode=debuginfo-gdb+lldb (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-18T06:50:26.0340294Z 
2019-10-18T06:50:26.0340609Z running 123 tests
2019-10-18T06:50:49.1154370Z .iiiii...i.....i..i...i..i.i.i..i.ii..i.i.....i..i....ii..........iiii..........i...ii...i.......ii. 100/123
2019-10-18T06:50:53.5962103Z i.i.i......iii.i.....ii
2019-10-18T06:50:53.5963216Z 
2019-10-18T06:50:53.5967940Z  finished in 27.724
2019-10-18T06:50:53.5977394Z Uplifting stage1 rustc (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-18T06:50:53.5977799Z Copying stage2 rustc from stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu / x86_64-unknown-linux-gnu)
---
2019-10-18T07:03:22.8561347Z 
2019-10-18T07:03:22.8565110Z    Doc-tests core
2019-10-18T07:03:27.7077135Z 
2019-10-18T07:03:27.7078021Z running 2406 tests
2019-10-18T07:03:38.4310831Z ......iiiii......................................................................................... 100/2406
2019-10-18T07:03:48.9070076Z ................................................................................ii.................. 200/2406
2019-10-18T07:04:13.5289832Z ..i................................................................................................. 400/2406
2019-10-18T07:04:13.5289832Z ..i................................................................................................. 400/2406
2019-10-18T07:04:23.6309574Z .................................................i..i.................iiii.......................... 500/2406
2019-10-18T07:04:43.3089745Z .................................................................................................... 700/2406
2019-10-18T07:04:53.6072845Z .................................................................................................... 800/2406
2019-10-18T07:05:03.7297995Z .................................................................................................... 900/2406
2019-10-18T07:05:13.8228849Z .................................................................................................... 1000/2406
---
2019-10-18T07:09:16.3359495Z 
2019-10-18T07:09:16.3360536Z running 997 tests
2019-10-18T07:09:37.8450461Z i................................................................................................... 100/997
2019-10-18T07:09:50.3645286Z .................................................................................................... 200/997
2019-10-18T07:09:57.5857219Z ...................iii......i......i...i......i..................................................... 300/997
2019-10-18T07:10:03.2003122Z .................................................................................................... 400/997
2019-10-18T07:10:10.7870807Z ........................................i..i.................................ii..................... 500/997
2019-10-18T07:10:25.4980394Z .................................................................................................... 700/997
2019-10-18T07:10:25.4980394Z .................................................................................................... 700/997
2019-10-18T07:10:33.1194147Z .......................iiii......................................................................... 800/997
2019-10-18T07:10:48.5049872Z .................................................................................................... 900/997
2019-10-18T07:10:56.2157509Z .............................................iiii................................................
2019-10-18T07:10:56.2159105Z 
2019-10-18T07:10:56.2223341Z  finished in 190.079
2019-10-18T07:10:56.2239947Z Testing term stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-10-18T07:10:56.4237330Z    Compiling term v0.0.0 (/checkout/src/libterm)
---
2019-10-18T07:25:42.2164866Z Rustbook (x86_64-unknown-linux-gnu) - edition-guide
2019-10-18T07:25:42.5577208Z Building stage0 tool linkchecker (x86_64-unknown-linux-gnu)
2019-10-18T07:25:42.7208091Z    Compiling linkchecker v0.1.0 (/checkout/src/tools/linkchecker)
2019-10-18T07:25:45.1232637Z     Finished release [optimized] target(s) in 2.45s
2019-10-18T07:25:47.0021278Z std/keyword.break.html:53: broken link - expressions/loop-expr.html
2019-10-18T07:25:52.9772825Z thread 'main' panicked at 'found some broken links', src/tools/linkchecker/main.rs:41:9
2019-10-18T07:25:52.9787068Z 
2019-10-18T07:25:52.9787420Z 
2019-10-18T07:25:52.9788194Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/linkchecker" "/checkout/obj/build/x86_64-unknown-linux-gnu/doc"
2019-10-18T07:25:52.9788593Z expected success, got: exit code: 101
---
2019-10-18T07:25:52.9863441Z   local time: Fri Oct 18 07:25:52 UTC 2019
2019-10-18T07:25:53.0587442Z   network time: Fri, 18 Oct 2019 07:25:53 GMT
2019-10-18T07:25:53.0590525Z == end clock drift check ==
2019-10-18T07:25:55.7110883Z 
2019-10-18T07:25:55.7247391Z ##[error]Bash exited with code '1'.
2019-10-18T07:25:55.7297214Z ##[section]Starting: Checkout
2019-10-18T07:25:55.7298920Z ==============================================================================
2019-10-18T07:25:55.7298989Z Task         : Get sources
2019-10-18T07:25:55.7299040Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

dorfsmay commented:
Fixed the link issue + rebase + forced push.

Dylan-DPC commented:
r? @Dylan-DPC

kinnison commented on src/libstd/keyword_docs.rs:
```diff
@@ -33,9 +33,71 @@ mod as_keyword { }
+///
+/// When associated with `loop`, but not with any other kind of loop expression,
+/// `break` can return a value.  When no value is specified, `break;` returns `()`.
+/// Every `break` within a loop must return the same type.
```
I've not done this as a suggestion because that doesn't work with the beta multi-line comment stuff.

I kinda feel like `break` doesn't "return" a value per-se, it's given a value for the `loop` to return.  As such, perhaps the following might be better wording?

```
/// When associated with a `loop` expression, but not with any other iterative expression,
/// `break` can be given a value to return.  When no value is given, `break;` returns `()`.
/// Every `break` within a `loop` must be given values of the same type.
```

dorfsmay commented:
Agreed. I tried to simplify too much. I have changed it, let me know what you think.

Also, I added a commit as per guideline, but can rebase if it is better.

Dylan-DPC commented:
@bors: r+ rollup

bors commented:
:pushpin: Commit 11214a63125efc187ef0c3acd90f8f8cf7c93dc6 has been approved by `Dylan-DPC`

<!-- @bors r=Dylan-DPC 11214a63125efc187ef0c3acd90f8f8cf7c93dc6 -->
<!-- homu: {"type":"Approved","sha":"11214a63125efc187ef0c3acd90f8f8cf7c93dc6","approver":"Dylan-DPC"} -->
